accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+```json
+{
+  "sdk": {
+    "version": "7.0.0"
+  }
+}
+```
+
+The primary value of the multi-level lookup is described by the following example scenario:
+
+- Build and run a .NET 6 app with a private .NET 7 SDK.
+- The .NET 7 SDK is used to build the app.
+- Multi-level lookup is used to find a .NET 6 runtime that is installed globally but not in the private location.
+- The app runs (via the globally installed .NET 6).
+
+It is simpler to install the .NET 6 runtime to the private location, resulting in much higher certainty that the desired versions are used. Multi-level lookup is best thought of as a [YOLO feature](https://en.wikipedia.org/wiki/YOLO_(aphorism)) and should have never been enabled by default (if it was needed at all).
+
+### Convention location
+
+> Customer value: This feature satisfies the "Dev inner loop consistency with lab builds" customer scenario.
+
+Note: This feature comes with significant cost and implementation complexity. We haven't decided if/when we will build it. We need more feedback to validate that it is valuable.
+
+.NET Team build scripts implement a convention where a private .NET SDK is installed into the `.dotnet` directory at the repo root. The version is controlled via the [version in the `global.json` at repo root](https://github.com/dotnet/iot/blob/79b9c35b6b575917ab841cd1b499a840c6f15bd1/global.json#L11). Currently, this pattern is just a convention, which `dotnet` doesn't honor. You have to add the `.dotnet` directory to the `PATH` if you want to be able to type `dotnet` and use the private .NET installation. We will teach `dotnet` to recognize this pattern, and instead proxy the launch of the global `dotnet` to the private `.dotnet` location. One can think of this as the opposite behavior of multi-level lookup.
+
+There are two aspects we need to consider for this experience:
+
+- Repos should be able to require the use of a private install.
+- This feature needs to be opt-in, per the principles stated earlier, and to ensure that the workspace is explicitly trusted by the user (similar to [VS Code Workspace Trust](https://code.visualstudio.com/docs/editor/workspace-trust)).
+
+"Convention location" characteristics:
+
+- Only supported for the `.dotnet` directory (not configurable to something else).
+- The directory must contain an SDK installation, otherwise error. The user will be directed to the [dotnet-install script](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script) help page.
+- Can be in current directory or (grand-*)parent (same probing logic as `global.json`).
```

Comment:
(grand-*) - expand here. 

If an attacker decides to drop a config file in drive root that'd be very ugly. I had suggested this directory only, no inheritance from directories above

Reply:
If it was current directory only, it can be trivially achieved today with a small .cmd/.sh file. We have a `dotnet` script like that in number of dotnet repos that forwards to the private SDK (or global SDK if it has good version).

Reply:
The UX doesn't work if it is always "this directory". Also, I don't think there is a security reason for that. I don't see it.